### PR TITLE
[DEVOPS-695] nix-darwin config for hydra build slaves

### PR DESCRIPTION
**Pull request by rvl**
Declarative configuration of macOS Hydra build slaves.

### CODE REVIEW

File: lib.nix [hunk 1]
```diff
@@ -77,4 +77,13 @@ in lib // (rec {
     philippKey
     vasilisKey
   ]);
+
+  buildSlaveKeys = with ssh-keys; {
+    macos = devOpsKeys ++ [
+      hydraBuildFarmKey
+      rodneyRemoteBuildKey
```
Comment (cleverca22) [inline lib.nix]: `rodneyRemoteBuildKey` for testing?

File: modules/hydra-slave.nix [hunk 2]
```diff
@@ -16,7 +16,7 @@ in
 
   environment.systemPackages = [ iohk-pkgs.iohk-ops ];
 
-  users.extraUsers.root.openssh.authorizedKeys.keys = pkgs.lib.singleton ''
-    command="nice -n20 ${pkgs.utillinux}/bin/flock -s /var/lock/lab nix-store --serve --write" ${pkgs.lib.readFile ./../static/id_buildfarm.pub}
-  '';
+  users.extraUsers.root.openssh.authorizedKeys.keys = map (key: ''
+    command="nice -n20 ${pkgs.utillinux}/bin/flock -s /var/lock/lab nix-store --serve --write" ${key}
```
Comment (cleverca22) [inline modules/hydra-slave.nix]: i think the flock part should go, its forcing the slave to only run 1 job at once, and hydra is configured to run 4 at once on each slave

File: nix-darwin/roles/buildkite-agent.nix [hunk 3]
```diff
@@ -8,6 +8,6 @@
   ];
 
   services = {
-    dd-agent.tags = [" group:buildkite-agents"];
+    dd-agent.tags = [" group:buildkite-agents" "group:macos"];
```
Comment (cleverca22) [inline nix-darwin/roles/buildkite-agent.nix]: we should be able to remove the redundant space before group, that was to fix a buildkite module bug

File: nix-darwin/roles/hydra-slave.nix [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+{
+  imports = [
+    ../modules/basics.nix
+    ../modules/datadog.nix
+    ../modules/hydra-slave.nix
+  ];
+
+  services = {
+   dd-agent.tags = [" group:hydra-and-slaves" "group:macos"];
```
Comment (cleverca22) [inline nix-darwin/roles/hydra-slave.nix]: redundant space again in the string

File: lib.nix (hunk 1 above)
Comment (rvl) [reply lib.nix]: This is because in my config I want to use a separate ssh key for nix remote builders. Is this OK?

**Comment by rvl** [comment]
Added review fixes and updated readme.

**Comment by rvl** [comment]
Rebased and squashed on develop which should fix the buildkite CI status.

**Pull request merged**